tests/agentic_system/tasks/test_dataloader.py

```diff
@@ -0,0 +1,262 @@
+import pytest
+import pandas as pd
+import numpy as np
+
+from dspy_litl_agentic_system.tasks.prism_lookup import PrismKey, PrismLookup
+
+
+@pytest.fixture
+def sample_data():
+    """Create sample PRISM data for testing."""
+    return pd.DataFrame({
+        'drug': ['DrugA', 'DrugB', 'DrugC', ' DrugD ', 'DrugA'],
+        'cell_line': ['CellX', 'CellY', 'CellZ', 'CellX', 'CellY'],
+        'ic50': [1.5, 2.3, 0.8, 3.1, 1.9],
+        'auc': [0.75, 0.82, 0.65, 0.91, 0.78],
+        'viability': [0.25, 0.18, 0.35, 0.09, 0.22]
+    })
+
+
+@pytest.fixture
+def sample_lookup(sample_data):
+    """Create a PrismLookup instance with sample data."""
+    return PrismLookup(sample_data)
+
+
+@pytest.fixture
+def casefold_lookup(sample_data):
+    """Create a PrismLookup instance with casefold=True."""
+    return PrismLookup(sample_data, casefold=True)
+
+
+class TestPrismKey:
+    """Test cases for PrismKey class."""
+    
+    def test_prism_key_creation(self):
+        key = PrismKey("DrugA", "CellX")
+        assert key.drug == "DrugA"
+        assert key.cell == "CellX"
+    
+    def test_prism_key_immutable(self):
+        key = PrismKey("DrugA", "CellX")
+        with pytest.raises(AttributeError):
+            key.drug = "DrugB"
+    
```

Comment:
Consider using [`pytest` parametrization](https://docs.pytest.org/en/stable/example/parametrize.html) to help build multiple test cases along with their expectations as a preamble for one test function/method. It might seem like a hassle early on but generally saves time and reduces the complexity of knowing which tests capture what cases.